ext/gd/config.m4

```diff
@@ -138,15 +138,56 @@ AC_DEFUN([PHP_GD_JISX0208],[
   fi
 ])
 
+
+dnl Build and run a program to determine if GD has support for the given
+dnl format. The sole argument is the proper-noun-capitalized name of the
+dnl format -- basically the word Foo in gdImageCreateFromFoo -- such as
+dnl Png. If support for format Foo exists, HAVE_GD_FOO will be defined
+dnl to 1. The reason for this charade is that gd defines "junk" versions
+dnl of each gdImageCreateFromFoo function even when it does not support
+dnl the Foo format. Those junk functions display a warning but eventually
+dnl return normally, making a simple link or run test insufficient.
+AC_DEFUN([PHP_GD_CHECK_FORMAT],[
+  old_LIBS="${LIBS}"
+  LIBS="${LIBS} ${GD_SHARED_LIBADD}"
+  AC_MSG_CHECKING([for working gdImageCreateFrom$1 in libgd])
+  AC_LANG_PUSH([C])
+  AC_RUN_IFELSE([AC_LANG_SOURCE([[
+#include <stdio.h>
+#include <unistd.h>
+#include <gd.h>
+
+/* A custom gdErrorMethod */
+void exit1(int priority, const char *format, va_list args) {
+  _exit(1);
+}
+
+/* Override the default gd_error_method with one that
+   actually causes the program to return an error. */
+int main(int argc, char** argv) {
+  FILE* f = NULL;
+  gdSetErrorMethod(exit1);
+  gdImagePtr p = gdImageCreateFrom$1(f);
+  return 0;
+} ]])],[
+    AC_MSG_RESULT([yes])
+    AC_DEFINE(HAVE_GD_[]m4_toupper($1), 1, [ ])
+  ],[
+    AC_MSG_RESULT([no])
+  ])
```

Comment:
```suggestion
  ],[
    AC_MSG_RESULT([no])
  ])
```

When cross-compiling:

```sh
./configure --host=aarch64-unknown-freebsd13.0 --build=x86_64 --enable-gd --with-external-gd
```

the AC_RUN_IF_ELSE, needs also a 3rd argument. Not that this even works at the moment in all cases, but let's add it anyway and we can simply disable it there because the [libgd](https://github.com/libgd/libgd/) seems to need some more patches.